### PR TITLE
Bug 1920629: templates/openshift/installer: Move installer work dir to /tmp/shared

### PR DESCRIPTION
**Pull request by mtnbikenc**
openshift-install should use a local scratch dir instead of dumping
everything to the artifact_dir.

### CODE REVIEW

**Comment by patrickdillon** [comment]
Looks like this solves the problem (there is no longer an `installer` dir with `bootstrap.ign`) and bootstrap completes.

/lgtm

**Comment by mtnbikenc** [comment]
Install failed and I don't know if that was due to changes in this PR or platform issues.
/test pj-rehearse

**Comment by mtnbikenc** [comment]
e2e-metal job passed.  Squashed commits.

**Comment by mtnbikenc** [comment]
/retest
```
Failing tests:

[sig-instrumentation] Prometheus when installed on the cluster should have important platform topology metrics [Suite:openshift/conformance/parallel]
[sig-instrumentation] Prometheus when installed on the cluster should start and expose a secured proxy and unsecured metrics [Suite:openshift/conformance/parallel]
[sig-instrumentation] Prometheus when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog and AlertmanagerReceiversNotConfigured [Early] [Suite:openshift/conformance/parallel]
```

**Comment by mtnbikenc** [comment]
/retest
```
Failing tests:

[sig-instrumentation] Prometheus when installed on the cluster should have important platform topology metrics [Suite:openshift/conformance/parallel]
[sig-instrumentation] Prometheus when installed on the cluster should start and expose a secured proxy and unsecured metrics [Suite:openshift/conformance/parallel]
[sig-instrumentation] Prometheus when installed on the cluster shouldn't report any alerts in firing state apart from Watchdog and AlertmanagerReceiversNotConfigured [Early] [Suite:openshift/conformance/parallel]
```

**Comment by mtnbikenc** [comment]
Test failures possibly related to https://bugzilla.redhat.com/show_bug.cgi?id=1871303

File: ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml [hunk 1]
```diff
@@ -540,11 +542,15 @@ objects:
...
-        openshift-install --dir=${ARTIFACT_DIR}/installer wait-for install-complete 2>&1 | grep --line-buffered -v password &
+        openshift-install --dir=${TMP_SHARED}/installer wait-for install-complete 2>&1 | grep --line-buffered -v password &
         wait "$!"
 
         # Password for the cluster gets leaked in the installer logs and hence removing them.
-        sed -i 's/password: .*/password: REDACTED"/g' ${ARTIFACT_DIR}/installer/.openshift_install.log
+        sed '
+          s/password: .*/password: REDACTED/;
+          s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+          s/UserData:.*,/UserData: REDACTED,/;
```
Comment (staebler) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml:552]: Should these substitutions be global?
```suggestion
          s/password: .*/password: REDACTED/g;
          s/X-Auth-Token.*/X-Auth-Token REDACTED/g;
          s/UserData:.*,/UserData: REDACTED,/g;
```

Comment (mtnbikenc) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml:552]: I don't see why it couldn't.  wking had suggested I update the original redaction with the code from the multi-stage installer step here, https://github.com/openshift/release/blob/9a726f0def12b3e735b38d74a95cd9a21c84720c/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L63-L67.  Upon searching the release repo I'm seeing instance where global is used so there is no consistency.  I'll update.

Comment (mtnbikenc) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml]: @staebler Updated.

**Review by staebler — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**